bayesian/utility/v2/stack_analyses.py

```diff
@@ -0,0 +1,243 @@
+_session = FuturesSession(max_workers=worker_count)
+
+
+class StackAnalyses():
+    """Implements stack analysis API.
+
+    Implements methods to support stack analyses post and get REST APIs calls.
+    """
+
+    def __init__(self, ecosystem=None, manifest=None, file_path=None, show_transitive=None):
+        """Initialize params to be used for ."""
+        self.ecosystem = ecosystem
+        self.manifest = manifest
+        self.file_path = file_path
+        self.show_transitive = show_transitive
+
+        # Private members used during post request.
+        self._new_request_id = None
+        self._request_date_str = None
+        self._manifest_file_info = None
+
+    def post_request(self):
+        """Make stack analyses POST request."""
+        # Make backbone request
+        deps = self._make_backbone_request()
+
+        # Finally save results in RDS and upon success return request id.
+        if self._save_request(deps) == 0:
+            return {
+                "status": "success",
+                "submitted_at": self._date_str,
+                "id": self._new_request_id
+            }
+
+    def validate_params(self):
+        """Validate mandatory post params of the request."""
+        # Manifest is mandatory and must be a string.
+        if self.manifest is None or not resolved_files_exist(self.manifest.filename):
+            error_message = "Error processing request. " \
+                            "Manifest is missing its value '{}' is " \
+                            "invalid / not supported".format(self.manifest)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # File path is mandatory and must be a string.
+        if self.file_path is None or not isinstance(self.file_path, str):
+            error_message = "Error processing request. " \
+                            "File path is missing / its value " \
+                            "'{}' is invalid".format(self.file_path)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Ecosystem  is mandatory and must be a string.
+        if self.ecosystem is None or not isinstance(self.ecosystem, str):
+            error_message = "Error processing request. " \
+                            "Ecosystem is missing / its value '{}' " \
+                            "is invalid".format(self.ecosystem)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Below ecosystem map tries to find the map entry for given ecosystem.
+        ecosystem_map = {
+            "node": "npm",
+            "python": "pypi",
+            "java": "maven"
+        }
+        # If given ecosystem is not found in above map, than uses the value passed in the request.
+        self.ecosystem = ecosystem_map.get(self.ecosystem, self.ecosystem)
```

Comment:
This map is not required, this is exactly what the util function is doing

Reply:
Can you point me to the function in util, which does this?

Reply:
Not required, I misread the code.